src/shared/components/query/CancerStudySelector.tsx

```diff
@@ -0,0 +1,267 @@
+		}
+
+		function setExpanded(node:Node, value:boolean):void {
+			dict_cancerTypeId_expanded[node.typeOfCancerId] = value;
+		}
+
+		let showStudiesInTree = false;
+
+		function getChildren(node:Node):Node[]|undefined {
+			if (!hasChildren(node))
+				return undefined;
+
+			let children:Node[] = dict_cancerTypeId_childCancerTypes[node.typeOfCancerId] || [];
+			if (!children.length && showStudiesInTree)
+				children = dict_cancerTypeId_childStudies[node.typeOfCancerId] || [];
+			return _.filter(children, filter);
+		}
+
+		let getContent = (node:Node) => {
+			if (!isCancerType(node))
+				return JSON.stringify(node);
+
+			let cancerType = node as TypeOfCancer;
+
+			let flatStudiesList = dict_cancerTypeId_flatStudiesList[cancerType.typeOfCancerId];
+			let selectedChildStudies = _.intersection(this.selectedStudies, flatStudiesList.map(study => study.cancerStudyIdentifier));
+			let checked = selectedChildStudies.length > 0;
+			let indeterminate = checked && selectedChildStudies.length != flatStudiesList.length;
+			return (
+				<Checkbox
+					selected={false}
+					onChange={this.getOnCheckHandler(cancerType)}
+					checked={checked}
+					inputRef={input => {
+						if (input)
+							input.indeterminate = indeterminate;
+					}}
+
+				>
+					{`${cancerType.name} (${flatStudiesList.length})`}
+				</Checkbox>
+			);
+		}
+
+    	return {
+    		rootNode: rootCancerType,
+    		isExpanded,
+    		setExpanded,
+    		getChildren,
+    		getContent,
+    	};
+    }
+
+    get selectedCancerType() {
+    	return this.props.selectedCancerType || this.state.selectedCancerType || CANCER_TYPE_ROOT;
+	}
+
+    get selectedStudies() {
+    	return this.props.selectedStudies || this.state.selectedStudies || [];
+	}
+
+	@memoizeWith(function() { return [this.props.cancerTypes, this.props.studies] })
+	getOnCheckHandler(node:Node) {
+		return (event:React.FormEvent) => {
+			let clickedStudies:string[];
+			let treeInfo = this.getTreeInfo();
+			if (isCancerType(node))
+			{
+				let cancerType = node as TypeOfCancer;
+				let flatStudiesList = treeInfo.dict_cancerTypeId_flatStudiesList[cancerType.typeOfCancerId];
+				clickedStudies = flatStudiesList.map(study => study.cancerStudyIdentifier);
+			}
+			else
+			{
+				let study = node as CancerStudy;
+		 		clickedStudies = [study.cancerStudyIdentifier];
+			}
+
+			let selectedStudies = this.selectedStudies;
+			let checked = (event.target as HTMLInputElement).checked;
+			if (checked)
+				selectedStudies = _.union(selectedStudies, clickedStudies);
+			else
+				selectedStudies = _.difference(selectedStudies, clickedStudies);
+
+			this.setState({selectedStudies});
+
+			if (this.props.onStateChange)
+				this.props.onStateChange({
+					selectedCancerType: this.selectedCancerType,
+					selectedStudies,
+				});
+		};
+	}
+
+    renderStudyName = (name:string, study:CancerStudy) => {
+    	let checked = !!_.find(this.selectedStudies, id => id == study.cancerStudyIdentifier);

```

Comment:
as above, the == enables type coercion.  I've always understood that === is preferable unless we actually want coercion.